### PR TITLE
Update Go

### PR DESCRIPTION
**Pull request by JakobGray**
Update go to v1.16. Move main.go to project root. Update dockerfiles.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Copyright Contributors to the Open Cluster Management project
 
-REGISTRY ?= quay.io/rhibmcollab
+REGISTRY ?= quay.io/open-cluster-management
```
Comment (zkayyali812) [inline Makefile:3]: Should we use `open-cluster-management` repo? For local development I usually use my personal quay repo. Last I recall, I dont think we have push access w/ our default accounts to `open-cluster-management` repo.

Comment (JakobGray) [reply Makefile:3]: I think we aren't technically supposed to be using rhibmcollab and wanted to match it with where our images are officially stored, with the expectation that anyone who wants to build our repo will push to their own registry. @nathanweatherly WDYT?

Comment (JakobGray) [reply Makefile:3]: I've reverted the name change

**Pull request merged**